queries/highlights.scm

```diff
@@ -1,132 +1,271 @@
-(comment) @comment
+; Includes
 
 [
-  "("
-  ")"
-  "{"
-  "}"
-  "["
-  "]"
-] @punctuation.bracket
+  "use"
+] @include
 
-[
-  ";"
-  "."
-  ","
-] @punctuation.delimiter
+; Keywords
 
 [
-  "if"
-  "ifdef"
-  "then"
-  "else"
-  "elseif"
-  "end"
-  "try"
-  "while"
-  "for"
-  "use"
-  "as"
-  "var"
-  "let"
-  "embed"
-  "fun"
-  "be"
-  "new"
+  "type"
   "actor"
   "class"
-  "struct"
   "primitive"
   "interface"
   "trait"
-  "type"
+  "struct"
+  "embed"
+  "let"
+  "var"
+  "be"
+  (compile_intrinsic)
+  "as"
+  "consume"
+  "object"
+  "where"
+] @keyword
+
+[
+  "fun"
+] @keyword.function
+
+[
+  "new"
+  "in"
+  "is"
+] @keyword.operator
+
+[
+  "return"
+] @keyword.return
+
+; Qualifiers
+
+[
   "iso"
   "trn"
   "ref"
   "val"
   "box"
   "tag"
+] @type.qualifier
+
+; Conditionals
+
+[
+  "if"
+  "ifdef"
+  "iftype"
+  "then"
+  "else"
+  "elseif"
+  "until"
+  "match"
+  "break"
+] @conditional
+
+(if_statement "end" @conditional)
+
+(iftype_statement "end" @conditional)
+
+(match_expression "end" @conditional)
+
+; Repeats
+
+[
+  "repeat"
+  "while"
+  "for"
+  "continue"
+  "do"
+] @repeat
+
+(do_block "end" @repeat)
+
+(repeat_statement "end" @repeat)
+
+; Exceptions
+
+[
+  "try"
+  "recover"
```

Comment:
`recover` is not used for recovering from errors, but to transform references to have another capability.
E.g. we can use it to turn a freshly created string from a `ref^` into an `iso^` capability:

```pony
let iso_str: String iso =
  recover iso
     String.create(5) .> append("hello")
  end
```

so it should maybe go underneath `keyword`.